pimd/pim_cmd.c

```diff
@@ -2369,6 +2371,19 @@ static void pim_show_upstream(struct pim_instance *pim, struct vty *vty,
 		pim_time_timer_to_hhmmss(join_timer, sizeof(join_timer),
 					 up->t_join_timer);
 
+
+		/*
+		 * if source/group passed filter based on that
+		 */
+		if (src_or_group) {
+			if (strcmp(src_or_group, src_str)
```

Comment:
Please consider changing the string to inaddr and compare it directly with  up->sg.src/grp? 
By that 
1. You can move this check to top of the for loop and avoid string manipulation  of unnecessary upstream entries.
2. numeric comparision will be more effective than strcmp
3. You can also validate the ip address string if it is in right format and throw error.

Reply:
sure will take care. Thnx for your time